app/src/androidTest/java/ch/epfl/cs311/wanderwave/ui/ProfileTest.kt

```diff
@@ -33,12 +35,15 @@ import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
+@HiltAndroidTest
 class ProfileTest : TestCase(kaspressoBuilder = Kaspresso.Builder.withComposeSupport()) {
 
   @get:Rule val composeTestRule = createAndroidComposeRule<TestActivity>()
 
   @get:Rule val mockkRule = MockKRule(this)
 
+  @get:Rule val hiltRule = HiltAndroidRule(this)
```

Comment:
Why do you need this every time ?

Reply:
To be able to use Hilt in the test at all, I had to change the TestRunner to a CustomTestRunner globally. For tests that use a specific activity like `createAndroidComposeRule<TestActivity>()` instead of just `createComposeRule()`, this requires the test to be annotated with @HiltAndroidTest and have the `HiltAndroidRule` as is the case here.
The whole thing is a bit of refactoring, but it all seems to be necessary for end-to-end tests to be possible.